icicle-compiler/src/Icicle/Runtime/Evaluator.hs

```diff
@@ -451,6 +502,67 @@ snapshot runtime maxsize stime input =
   in
     if Map.size both /= Map.size inputs then
       left $ RuntimeInputClusterMismatch (Map.keysSet missingInputs) (Map.keysSet missingClusters)
-    else
+    else do
       fmap (Output keys . Map.unions) . for (Map.elems both) $ \(cluster, column) ->
-        clusterSnapshot cluster maxsize stime column
+        snapshotCluster cluster maxsize stime column
+
+------------------------------------------------------------------------
+-- Skeleton for generated cluster state
+--
+-- typedef struct {
+--     /* runtime */
+--     anemone_mempool_t *mempool;
+--     iint_t             max_map_size;
+--
+--     /* input */
+--     itime_t   input_query_time;
+--     iint_t    input_count;
+--     ierror_t  *input_tombstone;
+--     iany_t    *input_melted0;
+--     iany_t    *input_melted1;
+--     iany_t    *input_melted2;
+--     ..
+--     iany_t    *input_meltedN;
+--     itime_t   *input_time;
+--
+--     /* kernel 0:0 outputs */
+--     iany_t     output0_melted0;
+--     iany_t     output0_melted1;
+--     ..
+--     iany_t     output0_meltedN;
+--
+--     /* kernel 0:1 outputs */
+--     iany_t     output1_melted0;
+--     iany_t     output1_melted1;
+--     ..
+--     iany_t     output1_meltedN;
+--
+--     /* kernel 0:0 flags */
+--     ibool_t has_flags_start_0_0;
+--     ibool_t has_0_0_melted0;
+--     ibool_t has_0_0_melted1;
+--     ..
+--     ibool_t has_0_0_meltedN;
+--     ibool_t has_flags_end_0_0;
+--
+--     /* kernel 0:1 flags */
+--     ibool_t has_flags_start_0_1;
+--     ibool_t has_0_1_melted0;
+--     ibool_t has_0_1_melted1;
+--     ..
+--     ibool_t has_0_1_meltedN;
+--     ibool_t has_flags_end_0_1;
+--
+--     /* kernel 0:0 resumables */
+--     ibuf_1_any_t res_0_0_melted0;
+--     ibuf_1_any_t res_0_0_melted1;
+--     ..
+--     ibuf_1_any_t res_0_0_meltedN;
+--
+--     /* kernel 0:1 resumables */
+--     ibuf_1_any_t res_0_1_melted0;
+--     ibuf_1_any_t res_0_1_melted1;
+--     ..
+--     ibuf_1_any_t res_0_1_meltedN;
+-- } cluster_0_t;
+--
```

Comment:
ok, scanned this but didn't look too closely, I'm assuming it's 👍 since icicle would explode violently if it doesn't